In [2]:
import numpy as np
import pandas as pd

In [3]:
df = pd.read_csv("ecommerce_sales.csv")

In [4]:
df.head()

,order_id,order_date,customer_id,product_category,region,quantity,unit_price,discount,payment_method,delivery_days,customer_rating,revenue
0,10001,1/1/2022,1102,Beauty,South,7,373.65,0.28,Wallet,10,4.7,1883.20
1,10002,1/2/2022,1435,Clothing,South,7,47.74,0.09,Card,6,3.9,304.10
2,10003,1/3/2022,1860,Beauty,East,3,311.28,0.31,COD,6,2.5,644.35
3,10004,1/4/2022,1270,Electronics,West,5,524.47,0.02,Wallet,6,1.6,2569.90
4,10005,1/5/2022,1106,Clothing,West,5,139.87,0.33,Wallet,4,4.9,468.56


In [5]:
df.tail()

,order_id,order_date,customer_id,product_category,region,quantity,unit_price,discount,payment_method,delivery_days,customer_rating,revenue
4995,14996,9/5/2035,1289,Clothing,West,2,371.74,0.27,COD,3,2.3,542.74
4996,14997,9/6/2035,1294,Beauty,North,3,553.67,0.14,COD,11,2.2,1428.47
4997,14998,9/7/2035,1450,Beauty,West,3,77.22,0.24,Wallet,6,3.4,176.06
4998,14999,9/8/2035,1903,Electronics,South,4,138.41,0.21,COD,4,2.8,437.38
4999,15000,9/9/2035,1214,Home,East,7,231.58,0.34,COD,5,3.9,1069.90


In [6]:
df.columns

Index(['order_id', 'order_date', 'customer_id', 'product_category', 'region',
       'quantity', 'unit_price', 'discount', 'payment_method', 'delivery_days',
       'customer_rating', 'revenue'],
      dtype='str')

In [7]:
df.shape

(5000, 12)

In [8]:
df.dtypes

order_id              int64
order_date              str
customer_id           int64
product_category        str
region                  str
quantity              int64
unit_price          float64
discount            float64
payment_method          str
delivery_days         int64
customer_rating     float64
revenue             float64
dtype: object

In [9]:
df.duplicated().sum()

np.int64(0)

In [10]:
df.describe()

,order_id,customer_id,quantity,unit_price,discount,delivery_days,customer_rating,revenue
count,5000.000000,5000.000000,5000.000000,5000.000000,5000.000000,5000.000000,5000.000000,5000.000000
mean,12500.500000,1505.701200,4.044800,308.418774,0.179984,6.118800,2.973980,1021.955148
std,1443.520003,290.836902,2.020398,169.259369,0.101404,3.153264,1.157722,825.584219
min,10001.000000,1000.000000,1.000000,15.150000,0.000000,1.000000,1.000000,11.210000
25%,11250.750000,1253.000000,2.000000,161.895000,0.090000,3.000000,2.000000,354.527500
50%,12500.500000,1510.000000,4.000000,309.890000,0.180000,6.000000,3.000000,796.650000
75%,13750.250000,1761.000000,6.000000,455.557500,0.270000,9.000000,4.000000,1515.690000
max,15000.000000,1999.000000,7.000000,599.960000,0.350000,11.000000,5.000000,4119.330000


In [11]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   order_id          5000 non-null   int64  
 1   order_date        5000 non-null   str    
 2   customer_id       5000 non-null   int64  
 3   product_category  5000 non-null   str    
 4   region            5000 non-null   str    
 5   quantity          5000 non-null   int64  
 6   unit_price        5000 non-null   float64
 7   discount          5000 non-null   float64
 8   payment_method    5000 non-null   str    
 9   delivery_days     5000 non-null   int64  
 10  customer_rating   5000 non-null   float64
 11  revenue           5000 non-null   float64
dtypes: float64(4), int64(4), str(4)
memory usage: 468.9 KB


In [12]:
df.isnull().sum()

order_id            0
order_date          0
customer_id         0
product_category    0
region              0
quantity            0
unit_price          0
discount            0
payment_method      0
delivery_days       0
customer_rating     0
revenue             0
dtype: int64

In [13]:
df["product_category"].value_counts()

product_category
Electronics    1777
Clothing       1531
Home            969
Beauty          723
Name: count, dtype: int64

In [14]:
df["region"].value_counts()

region
West     1289
North    1276
East     1227
South    1208
Name: count, dtype: int64

In [15]:
df["order_date"]= pd.to_datetime(df["order_date"])

In [16]:
df["payment_method"].value_counts()

payment_method
Card      2270
COD       1774
Wallet     956
Name: count, dtype: int64

In [17]:
df.groupby("product_category")["revenue"].sum()

product_category
Beauty          765860.88
Clothing       1531931.72
Electronics    1829899.22
Home            982083.92
Name: revenue, dtype: float64

In [18]:
df.groupby("product_category")["revenue"].mean()

product_category
Beauty         1059.281992
Clothing       1000.608570
Electronics    1029.768835
Home           1013.502497
Name: revenue, dtype: float64

In [19]:
df.groupby("product_category")["revenue"].max()

product_category
Beauty         4005.93
Clothing       3928.63
Electronics    4119.33
Home           3640.56
Name: revenue, dtype: float64

In [20]:
df.groupby("product_category")["revenue"].min()

product_category
Beauty         20.94
Clothing       11.21
Electronics    13.26
Home           13.01
Name: revenue, dtype: float64

In [21]:
df["payment_method"].value_counts()

payment_method
Card      2270
COD       1774
Wallet     956
Name: count, dtype: int64

In [22]:
df.groupby("payment_method")["revenue"].agg(["sum" ,"mean"])

,sum,mean
payment_method,,
COD,1788408.44,1008.122007
Card,2366248.20,1042.400088
Wallet,955119.10,999.078556


In [23]:
df["region"].value_counts()

region
West     1289
North    1276
East     1227
South    1208
Name: count, dtype: int64

In [24]:
df.groupby("region")["revenue"].agg(["sum","mean"])

,sum,mean
region,,
East,1236044.23,1007.371011
North,1281508.45,1004.316967
South,1246640.90,1031.987500
West,1345582.16,1043.896168


In [25]:
df.head()

,order_id,order_date,customer_id,product_category,region,quantity,unit_price,discount,payment_method,delivery_days,customer_rating,revenue
0,10001,2022-01-01,1102,Beauty,South,7,373.65,0.28,Wallet,10,4.7,1883.20
1,10002,2022-01-02,1435,Clothing,South,7,47.74,0.09,Card,6,3.9,304.10
2,10003,2022-01-03,1860,Beauty,East,3,311.28,0.31,COD,6,2.5,644.35
3,10004,2022-01-04,1270,Electronics,West,5,524.47,0.02,Wallet,6,1.6,2569.90
4,10005,2022-01-05,1106,Clothing,West,5,139.87,0.33,Wallet,4,4.9,468.56


In [26]:
df.groupby("product_category")["revenue"].agg(["sum", "mean" , "min" , "max"])

,sum,mean,min,max
product_category,,,,
Beauty,765860.88,1059.281992,20.94,4005.93
Clothing,1531931.72,1000.608570,11.21,3928.63
Electronics,1829899.22,1029.768835,13.26,4119.33
Home,982083.92,1013.502497,13.01,3640.56


In [27]:
df["product_category"].value_counts()

product_category
Electronics    1777
Clothing       1531
Home            969
Beauty          723
Name: count, dtype: int64

In [28]:
df["order_year"] = df["order_date"].dt.year

In [29]:
df["order_year"].value_counts()

order_year
2024    366
2028    366
2032    366
2022    365
2023    365
2025    365
2026    365
2027    365
2029    365
2030    365
2031    365
2033    365
2034    365
2035    252
Name: count, dtype: int64

In [30]:
df.groupby("order_year")["revenue"].agg(["sum"])

,sum
order_year,
2022,371824.30
2023,365915.79
2024,363209.13
2025,349802.33
2026,386369.69
2027,386151.91
2028,376844.80
2029,359239.24
2030,387884.46


In [31]:
working_df = df.copy()

In [32]:
working_df = df[df["order_year"] <= 2026]

In [33]:
working_df.shape

(1826, 13)

In [34]:
np.mean(working_df["revenue"])


np.float64(1006.0904928806135)

In [35]:
np.median(working_df["revenue"])

np.float64(783.5699999999999)

In [36]:
np.std(working_df["revenue"])

np.float64(821.8170609053799)

In [37]:
np.max(working_df["revenue"])

np.float64(4119.33)

In [38]:
np.min(working_df["revenue"])

np.float64(12.83)

In [39]:
np.argmax(working_df["revenue"])

np.int64(809)

In [40]:
max_pos = np.argmax(working_df["revenue"])
working_df.iloc[max_pos]

order_id                          10810
order_date          2024-03-20 00:00:00
customer_id                        1890
product_category            Electronics
region                            South
quantity                              7
unit_price                       594.42
discount                           0.01
payment_method                     Card
delivery_days                         4
customer_rating                     4.9
revenue                         4119.33
order_year                         2024
Name: 809, dtype: object

In [41]:
min_pos = np.argmin(working_df["revenue"])
working_df.iloc[min_pos]

order_id                          10565
order_date          2023-07-19 00:00:00
customer_id                        1341
product_category               Clothing
region                            North
quantity                              1
unit_price                        19.74
discount                           0.35
payment_method                      COD
delivery_days                         8
customer_rating                     4.3
revenue                           12.83
order_year                         2023
Name: 564, dtype: object

In [42]:
np.max(working_df["revenue"]) - np.min(working_df["revenue"])

np.float64(4106.5)

## Saving Files As Excel & CSV


In [43]:
working_df.to_csv("Finalized_Dataset.csv" , index=False)

In [44]:
working_df.to_excel("Finalized_Dataset.xlsx" , index=False)

# 📊 E-Commerce Sales Data Analysis — Final Report

## 1. Project Overview

This project analyzes an e-commerce sales dataset using **Python, Pandas, and NumPy**.

The main objectives were to:

* Understand the structure of the dataset
* Check data quality
* Analyze product categories
* Analyze payment methods
* Analyze regional performance
* Analyze revenue
* Analyze order dates
* Apply NumPy for numerical analysis
* Extract useful business insights from the data

---

## 2. Dataset Overview

After inspecting the dataset, I analyzed **1826 rows and 13 columns** in the working dataset.

The dataset contains information related to:

* Orders
* Product categories
* Revenue
* Payment methods
* Regions
* Order dates

The `order_date` column was in string format converted into a proper datetime format so that year/month-based analysis could be performed.

---

## 3. Data Quality Analysis

I checked the dataset for common data-quality problems.

### Checks performed

* Missing values
* Duplicate rows
* Data types
* Inconsistent categorical values
* Suspicious numerical values
* Date values

The dataset did not show major data-quality problems in the areas checked.

During date analysis, I noticed that the original dataset contained dates extending beyond 2026. For this analysis, I created a working dataset containing dates up to 2026 while keeping the original dataset unchanged.

---

## 4. Product Category Analysis

I analyzed product categories using order counts and revenue statistics.

### Key findings

* **Electronics** generated the highest total revenue.
* Total revenue from Electronics was approximately **₹18,29,899.22**.
* **Beauty** had the highest average revenue per order, approximately **₹1,059.28**.
* Electronics also had the highest individual revenue value, approximately **₹4,119.33**.
* Beauty had the lowest total revenue among the categories.

### Important observation

The category with the highest total revenue was not necessarily the category with the highest average revenue.

Therefore, "best category" depends on the metric being considered.

**Electronics was the best category based on total revenue, while Beauty was the best based on average revenue per order.**

---

## 5. Payment Method Analysis

I analyzed the different payment methods using order counts and revenue.

### Key findings

* **Card** was the most frequently used payment method.
* **Card** also generated the highest total revenue.
* Therefore, the most frequently used payment method was also the highest-revenue payment method.

### Insight

This indicates that Card was both the most popular payment method and the strongest payment method in terms of total revenue in this dataset.

---

## 6. Regional Analysis

I analyzed order volume and revenue across different regions.

### Key findings

* **West** had the highest number of orders.
* **West** generated the highest total revenue.
* **West** also had the highest average revenue.
* The region with the most orders was also the region with the highest revenue.

### Insight

West was the strongest-performing region across the main metrics analyzed.

---

## 7. Date/Time Analysis

The `order_date` column was converted to datetime and the year was extracted for analysis.

### Key findings

* **2024, 2028, and 2032** had the highest number of orders, with a similar/high order count.
* **2035** had the lowest number of orders.
* **2033** generated the highest total revenue, with revenue exceeding **₹40 lakh**.

### Insight

The year with the highest number of orders was not necessarily the year with the highest total revenue.

This shows that **order volume and revenue are different metrics** and should be analyzed separately.

we use the filtering methods to filter/Remove upcoming years 
---

## 8. NumPy Analysis

NumPy was used to perform numerical analysis on the revenue column.

The following statistics were calculated:

* Mean revenue
* Median revenue
* Standard deviation
* Minimum revenue
* Maximum revenue
* Revenue range
* Position of maximum revenue
* Position of minimum revenue

These statistics helped understand the distribution and variation of revenue values in the dataset.

### NumPy functions used

```python
np.mean()
np.median()
np.std()
np.min()
np.max()
np.argmax()
np.argmin()
```

---

## 9. Overall Business Insights

Based on the analysis, the following major insights were identified:

1. **Electronics generated the highest total revenue** among the product categories.
2. **Beauty had the highest average revenue per order**, despite having the lowest total revenue.
3. **Card was the most-used payment method** and also generated the highest total revenue.
4. **West was the strongest region**, having the highest order count, total revenue, and average revenue.
5. The year with the highest order volume was not necessarily the year with the highest revenue.
6. **2033 generated the highest yearly revenue**, exceeding ₹40 lakh.
7. Different metrics can produce different conclusions, so business performance should not be judged using only one metric.
8. We filter Upcoming Years / Remove years which are **2026** This Helps to Removes rows which are before it was 5000 rows of data after it becomes only approximately ~1900 rows . which don't affected orginal dataset


---

## 10. Conclusion

This project helped demonstrate how **Pandas and NumPy can be used together for real-world data analysis**.

Pandas was mainly used for:

* Data inspection
* Filtering
* Data cleaning
* Grouping
* Aggregation
* Date handling

NumPy was mainly used for:

* Numerical calculations
* Statistical analysis
* Finding minimum and maximum positions
* Understanding revenue variation

The main lesson from this project is that data analysis is not only about writing commands. It involves **asking questions, selecting the appropriate operation, analyzing the results, and explaining what those results mean**.
